### ISSUE TITLE
Manipulating SMBIOS system-serial-number

### ISSUE THREAD

**Issue by syncopsta** [cloud-hypervisor/cloud-hypervisor#4002]
I would find it useful if you could change the system-serial-number via cloud-hypervisor.
The QEMU equivalent is the option `-smbios type=1,serial= `

Especially for the use of cloud-init could it be handy, so that an external meta-data service can be requested.

List with the mappings: https://gist.github.com/smoser/290f74c256c89cb3f3bd434a27b9f64c
Information about the Cloud-Init NoCloud-Net Datasource: https://cloudinit.readthedocs.io/en/latest/topics/datasources/nocloud.html

**Comment by sboeuf** [comment]
@rbradford what do you think? We could use the platform option to add an extra parameter like this.

**Comment by rbradford** [comment]
Yes, adding it to `--platform` would be appropriate and the cloud-init use case seems reasonable.

**Comment by sboeuf** [comment]
@syncopsta we've merged the support for it. Please have a try using the latest `main` branch from Cloud Hypervisor.

**Comment by iggy** [comment]
If you ever want to expand the supported smbios fields, I found this useful table.

https://gist.github.com/smoser/290f74c256c89cb3f3bd434a27b9f64c
